### PR TITLE
Fix item wrongly appears selected when reordering selected item in LV

### PR DESCRIPTION
**Pull request by dr1rrb**
fixes https://github.com/unoplatform/nventive-private/issues/362
fixes https://github.com/unoplatform/uno/issues/4741

## Bugfix
When we re-order a selected item in a LV, the item above or below drop location might appears as selected.

## What is the current behavior?
When the moved item is being drop, we are doing remove+add which drives the LV to wrongly updates internal indices, which drives the selected state to be applied on the wrong item.

## What is the new behavior?
* On collection add/remove we are computing indices properly
* On Android, while reordering an item we no longer rely on the non-yet-updated native collection of items.

## PR Checklist
- [ ] ~~Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)~~
- [x] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.Foundation/Collections/VectorChangedEventArgs.cs [hunk 1]
```diff
@@ -1,19 +1,31 @@
-﻿using System;
+﻿#nullable enable
+
+using System;
 using System.Collections.Generic;
+using System.Collections.Specialized;
 using System.Text;
+using Android.Media;
```
Comment (jeromelaban) [inline src/Uno.Foundation/Collections/VectorChangedEventArgs.cs:7]: Not supposed to be here :)

File: src/SamplesApp/SamplesApp.UITests/TestFramework/IsExternalInit.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Linq;
+
+namespace System.Runtime.CompilerServices;
+
+#if !NET5_0 && !NET6_0
```
Comment (jeromelaban) [inline src/SamplesApp/SamplesApp.UITests/TestFramework/IsExternalInit.cs:6]:
```suggestion
#if !NET5_0 && !NET6_0_OR_GREATER
```

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs [hunk 3]
```diff
@@ -28,6 +28,8 @@ partial class ListViewBase
 		public event DragItemsStartingEventHandler DragItemsStarting;
 		public event TypedEventHandler<ListViewBase, DragItemsCompletedEventArgs> DragItemsCompleted;
 
+		protected bool _isProcessingReorder;
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs:31]: Does it need to be protected?

```suggestion
		private protected bool _isProcessingReorder;
```

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.Android.cs [hunk 4]
```diff
@@ -163,6 +163,14 @@ private ContentControl ContainerFromGroupIndex(int groupIndex)
 
 		internal override int IndexFromContainerInner(DependencyObject container)
 		{
+			if (_isProcessingReorder)
+			{
+				// When we process a re-ordering, the "attached" indexes (IndexForItemContainerProperty) are known to be valid,
+				// but the native (NativePanel) is not (moved item is still at it's original location / index).
+
+				return base.IndexFromContainerInner(container);
```
Comment (carldebilly) [inline src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.Android.cs:171]: Clever ;-)

**Review by carldebilly — APPROVED** [review]

File: src/SamplesApp/SamplesApp.UITests/TestFramework/IsExternalInit.cs (hunk 2 above)
Comment (dr1rrb) [reply src/SamplesApp/SamplesApp.UITests/TestFramework/IsExternalInit.cs:6]: Good point, will fix in another PR

File: src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs (hunk 3 above)
Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Controls/ListViewBase/ListViewBase.DragDrop.cs:31]: Your right ... I moved it without changing the modifier, will fix in another PR

**Pull request merged**